openmc/model/funcs.py

```diff
@@ -0,0 +1,578 @@
+from collections.abc import Iterable
+from math import sqrt
+from numbers import Real
+from functools import partial
+from warnings import warn
+from operator import attrgetter
+
+from openmc import (
+    XPlane, YPlane, Plane, ZCylinder, Quadric, Cylinder, XCylinder,
+    YCylinder, Material, Universe, Cell)
+from openmc.checkvalue import (
+    check_type, check_value, check_length, check_less_than,
+    check_iterable_type)
+import openmc.data
+
+
+def borated_water(boron_ppm, temperature=293., pressure=0.1013, temp_unit='K',
+                  press_unit='MPa', density=None, **kwargs):
+    """Return a Material with the composition of boron dissolved in water.
+
+    The water density can be determined from a temperature and pressure, or it
+    can be set directly.
+
+    The concentration of boron has no effect on the stoichiometric ratio of H
+    and O---they are fixed at 2-1.
+
+    Parameters
+    ----------
+    boron_ppm : float
+        The weight fraction in parts-per-million of elemental boron in the
+        water.
+    temperature : float
+        Temperature in [K] used to compute water density.
+    pressure : float
+        Pressure in [MPa] used to compute water density.
+    temp_unit : {'K', 'C', 'F'}
+        The units used for the `temperature` argument.
+    press_unit : {'MPa', 'psi'}
+        The units used for the `pressure` argument.
+    density : float
+        Water density in [g / cm^3].  If specified, this value overrides the
+        temperature and pressure arguments.
+    **kwargs
+        All keyword arguments are passed to the created Material object.
+
+    Returns
+    -------
+    openmc.Material
+
+    """
+    # Perform any necessary unit conversions.
+    check_value('temperature unit', temp_unit, ('K', 'C', 'F'))
+    if temp_unit == 'K':
+        T = temperature
+    elif temp_unit == 'C':
+        T = temperature + 273.15
```

Comment:
This would be a great place for a constant. Boiling point of water in Kelvin and any other reference points in temperatures would make sense as a constant.